x-pack/libbeat/management/api/convert.go

```diff
@@ -0,0 +1,81 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package api
+
+import (
+	"fmt"
+	"strings"
+)
+
+type convertFunc func(map[string]interface{}) (map[string]interface{}, error)
+
+var mapper = map[string]convertFunc{
+	".inputs":  noopConvert,
+	".modules": convertMultiple,
+	"output":   convertSingle,
+}
+
+var errSubTypeNotFound = fmt.Errorf("'%s' key not found", subTypeKey)
+
+var (
+	subTypeKey = "_sub_type"
+	moduleKey  = "module"
+)
+
+func selectConverter(t string) convertFunc {
+	c := noopConvert
+	for k, v := range mapper {
+		if strings.Index(t, k) > -1 {
+			c = v
+			return c
+		}
+	}
+	return c
```

Comment:
given the body and early return we don't really need 'c' :)

Reply:
Sorry, wasnt' clear. I was wondering if we need the variable at all:

```
for k, v := range mapper {
  if strings.Contains(t, k) {
    return v
  }
}
return noopConverter // no match
```

Reply:
Oh yeah thats correct, I've made the change.